dwave/plugins/torch/nn.py

```diff
@@ -0,0 +1,118 @@
+import inspect
+from functools import wraps
+from types import MappingProxyType
+
+import torch
+from torch import nn
+
+
+def store_config(fn):
+    @wraps(fn)
+    def wrapper(self, *args, **kwargs):
+        # Get signature of function and match the arguments with their names
+        sig = inspect.signature(fn)
+        bound = sig.bind(self, *args, **kwargs)
+        # Use default values if the args/kwargs were not supplied
+        bound.apply_defaults()
+        config = {k: v for k, v in bound.arguments.items() if k != 'self'}
+        config['module_name'] = bound.args[0].__class__.__name__
+        self.config = MappingProxyType(config)
+        fn(self, *args, **kwargs)
+    return wrapper
+
+
+class Identity(nn.Module):
+
+    @store_config
+    def __init__(self):
+        """An identity module.
+
+        This module is useful for handling cases where a neural network module is expected, but no
+        effect is desired."""
+        super().__init__()
+
+    def forward(self, x) -> torch.Tensor:
+        """Input
+
+        Args:
+            x (torch.Tensor): The input and the output.
+
+        Returns:
+            torch.Tensor: The input and the output.
+        """
+        return x
+
+
+class SkipLinear(nn.Module):
+
+    @store_config
+    def __init__(self, din, dout) -> None:
+        """Applies a linear transformation to the incoming data: :math:`y = xA^T`.
+
+        This module is identity when `din == dout`, otherwise, it is a linear transformation, i.e.,
```

Comment:
Is this what makes it a "Skip" linear instead of a plain linear module? I find this very counter-intuitive: that having the same output dimension performs an identity transformation instead of a linear transformation (like an endomorphism).